### PR TITLE
Added allow_memory_growth flag in ddsp_run

### PR DESCRIPTION
**Pull request by andreykramer**

### CODE REVIEW

**Comment by andreykramer** [comment]
@googlebot I signed it!

File: ddsp/training/ddsp_run.py [hunk 1]
```diff
@@ -139,6 +142,8 @@ def main(unused_argv):
   # Training.
   if FLAGS.mode == 'train':
     strategy = train_util.get_strategy(tpu=FLAGS.tpu, gpus=FLAGS.gpu)
+    if FLAGS.allow_memory_growth:
+        train_util.allow_memory_growth()
```
Comment (jesseengel) [inline ddsp/training/ddsp_run.py]: 2 space indent by the google style guide.

File: ddsp/training/train_util.py [hunk 2]
```diff
@@ -57,6 +57,17 @@ def get_strategy(tpu='', gpus=None):
     strategy = tf.distribute.MirroredStrategy()
   return strategy
 
+def allow_memory_growth():
+  """Sets the GPUs to grow the memory usage as is needed by the process"""
+  gpus = tf.config.experimental.list_physical_devices('GPU')
+  if gpus:
+    try:
+      # Currently, memory growth needs to be the same across GPUs
```
Comment (jesseengel) [inline ddsp/training/train_util.py]: nit: period at the end of comments.

File: ddsp/training/train_util.py [hunk 3]
```diff
@@ -57,6 +57,17 @@ def get_strategy(tpu='', gpus=None):
...
+      for gpu in gpus:
+        tf.config.experimental.set_memory_growth(gpu, True)
+    except RuntimeError as e:
+      # Memory growth must be set before GPUs have been initialized
```
Comment (jesseengel) [inline ddsp/training/train_util.py]: nit: period at the end of comments.

File: ddsp/training/train_util.py [hunk 4]
```diff
@@ -57,6 +57,17 @@ def get_strategy(tpu='', gpus=None):
     strategy = tf.distribute.MirroredStrategy()
   return strategy
 
+def allow_memory_growth():
+  """Sets the GPUs to grow the memory usage as is needed by the process"""
```
Comment (jesseengel) [inline ddsp/training/train_util.py]: nit: period at the end of first line of documentation.

**Comment by andreykramer** [comment]
I will confirm that it works on monday, as I won't have access to a PC with an Nvidia graphics card until then. Thank you for the style corrections.

**Comment by jesseengel** [comment]
Friendly ping on this :), no rush (not critical) but just wanted to check in

**Comment by andreykramer** [comment]
> 
> 
> Friendly ping on this :), no rush (not critical) but just wanted to check in

Sorry for the delay, tomorrow I will give you an update on this

**Comment by andreykramer** [comment]
It doesn't seem to work right now as-is, and currently I don't have the resources available to me to invest time on trying to fix it. Eventually I believe I will, so I'm leaving this on hold.

**Comment by jesseengel** [comment]
Okay, sgtm

**Comment by andreykramer** [comment]
Hello, I've moved the allow_memory_growth function from train_util directly to ddsp_run and I'm calling it in the main before the mode is checked because it's necessary not only for train but also for 'eval' and 'sample'. Everything is working now.

**Comment by jesseengel** [comment]
Great! Looks like you still just have a lint error: C0301: 89 : Line too long (82/80) [line-too-long]

**Comment by andreykramer** [comment]
@googlebot I signed it!

**Comment by andreykramer** [comment]
@googlebot I fixed it.

**Comment by jesseengel** [comment]
Ah, seems like you need a CLA for andrey.kramer@voicemod.net as well

**Comment by andreykramer** [comment]
@googlebot I fixed it.

**Comment by andreykramer** [comment]
I signed it with that mail as well and the bot still doesn't seem happy, maybe it's because I accidentaly closed the PR
@googlebot I fixed it.

**Comment by jesseengel** [comment]
Thanks for the contribution!

**Pull request merged**